src/Kiota.Builder/Configuration/PluginAuthConfiguration.cs

```diff
@@ -44,4 +46,37 @@ internal Auth ToPluginManifestAuth()
             _ => throw new ArgumentOutOfRangeException(nameof(AuthType), $"Unknown plugin auth type '{AuthType}'")
         };
     }
+
+    /// <summary>
+    /// Constructs a PluginAuthConfiguration object from SecuritySchemeType and reference id.
+    /// </summary>
+    /// <param name="pluginAuthType">The SecuritySchemeType.</param>
+    /// <param name="pluginAuthRefId">The reference id.</param>
+    /// <returns>A PluginAuthConfiguration object.</returns>
+    /// <exception cref="ArgumentException">If the reference id is null or contains only whitespaces.</exception>
+    /// <exception cref="ArgumentOutOfRangeException">If the SecuritySchemeType is unknown.</exception>
+    public static PluginAuthConfiguration FromParameters(SecuritySchemeType? pluginAuthType, string pluginAuthRefId)
```

Comment:
nit: I'm wondering if this should be an extension method?

Reply:
Keeping it as an instance method for now as we have control of the class.